messaging/src/main/java/org/axonframework/commandhandling/gateway/DefaultCommandGateway.java

```diff
@@ -164,13 +188,28 @@ private RuntimeException asRuntime(Throwable e) {
      * The {@link CommandBus} is a <b>hard requirements</b> and as such should be provided.
      */
     public static class Builder extends AbstractCommandGateway.Builder {
+        private CommandCallback<Object, Object> commandCallback = new FailureLoggingCallback<>(logger);
 
         @Override
         public Builder commandBus(CommandBus commandBus) {
             super.commandBus(commandBus);
             return this;
         }
 
+        /**
+         * Set a {@link CommandCallback} on the command bus. This will be used as callback for all asynchronous
+         * commands that are sent.
+         * By default, the {@link FailureLoggingCallback} is used. This will log to the default logger on failure.
+         *
+         * @param commandCallback The {@link CommandCallback} to use for asynchronous commands
+         * @return the current Builder instance, for fluent interfacing
+         */
+        public Builder commandCallbackCustomizer(CommandCallback<Object, Object> commandCallback) {
```

Comment:
This method should be renamed to `commandCallback`:
```suggestion
        public Builder commandCallback(CommandCallback<Object, Object> commandCallback) {
```